### PR TITLE
non namespace resources api call fixed

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

non namespaced resources fixed for fetching and edit

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: client/k8s/application/Application.go [hunk 1]
```diff
@@ -65,15 +65,23 @@ type EventsResponse struct {
 	Events *apiv1.EventList `json:"events,omitempty"`
 }
 
+func contains(slice []string, item string) bool {
+	set := make(map[string]struct{}, len(slice))
```
Comment (manish-agrawal-ai) [inline client/k8s/application/Application.go:69]: Isn't it a better solution?

for _, singleItem := range slice {
		if singleItem == item {
			return true
		}
	}
return false

File: client/k8s/application/Application.go [hunk 2]
```diff
@@ -253,3 +262,16 @@ func ServerResourceForGroupVersionKind(discoveryClient discovery.DiscoveryInterf
 	}
 	return nil, errors.NewNotFound(schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind}, "")
 }
+
+func (impl K8sClientServiceImpl) getNonNamespacedResourceKind(discoveryClient *discovery.DiscoveryClient) []string {
```
Comment (manish-agrawal-ai) [inline client/k8s/application/Application.go:266]: We are returning only kinds, 
we should return gvks and check gvk (not only kind)

Comment (manish-agrawal-ai) [reply client/k8s/application/Application.go:266]: I can't see changes in ListEvents and GetPodLogs, is this intentional?

File: client/k8s/application/Application.go [hunk 3]
```diff
@@ -220,25 +228,26 @@ func (impl K8sClientServiceImpl) GetPodLogs(restConfig *rest.Config, request *K8
...
-		return nil, err
+		return nil, nil, err
 	}
 	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
 	if err != nil {
 		impl.logger.Errorw("error in getting k8s client", "err", err)
-		return nil, err
+		return nil, nil, err
 	}
 	apiResource, err := ServerResourceForGroupVersionKind(discoveryClient, resourceIdentifier.GroupVersionKind)
```
Comment (manish-agrawal-ai) [inline client/k8s/application/Application.go:243]: No need to call saparate API for this (getNonNamespacedResourceKind)

in apiResource, you get namespaced bool

**Review by manish-agrawal-ai — COMMENTED** [review]
check comments

File: client/k8s/application/Application.go (hunk 2 above)
Comment (vikramdevtron) [reply client/k8s/application/Application.go]: done, logical changes applied

File: client/k8s/application/Application.go (hunk 1 above)
Comment (vikramdevtron) [reply client/k8s/application/Application.go]: done, logical changes applied

**Pull request merged**